src/browserList.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DebugController, DebugControllerState, ReusedBrowser } from './reusedBrowser';
+import { TestModelCollection } from './testModel';
+import * as vscodeTypes from './vscodeTypes';
+
+export class BrowserList {
+  private _state = new Map<DebugController, { id: string; name: string; channel?: string; title: string }[]>();
```

Comment:
I don't quite like that `BrowserList` touches `DebugController`, which is an internal part of it. I'd prefer for `BrowserList` to listen to `ReusedBrowser.onBrowsersChanged` event instead, and encapsulate the `DebugController` inside. WDYT?

Reply:
I think it's good how it is. Let me give an outlook for what i'm planning in the upcoming PRs:

`ReusedBrowser` won't be the only thing that contributes Debug Controllers, but the MCP integration will also contribute them. So I want all browser-related actions to live on `BrowserList` instead of `ReusedBrowser`. For that, I need DebugController to become an internal part of `BrowserList`. `ReusedBrowser` will lose a lot of responsibility to `BrowserList`, all the inspection and closing and recording stuff will move over to `BrowserList`, and `ReusedBrowser` ends up being just about starting/stopping the browser server.